src/main/java/org/broadinstitute/hellbender/tools/walkers/variantutils/ValidateVariants.java

```diff
@@ -367,50 +449,185 @@ private void validateGVCFVariant(final VariantContext vc) {
         }
     }
 
-    private void applyValidationType(VariantContext vc, Allele reportedRefAllele, Allele observedRefAllele, Set<String> rsIDs, ValidationType t) {
+    private void applyValidationType(VariantContext vc, Allele reportedRefAllele, Allele observedRefAllele, Set<String> rsIDs) {
         // Note: VariantContext.validateRSIDs blows up on an empty list (but works fine with null).
         // The workaround is to not pass an empty list.
-        switch( t ) {
-            case ALL:
-                if(hasReference())
-                {
-                    if(!rsIDs.isEmpty())
-                    {
-                        vc.extraStrictValidation(reportedRefAllele, observedRefAllele, rsIDs);
-                    }
-                    else{
-                        vc.validateReferenceBases(reportedRefAllele, observedRefAllele);
-                        vc.validateAlternateAlleles();
-                        vc.validateChromosomeCounts();
-                    }
-                }
-                else{
-                    if (rsIDs.isEmpty())
-                    {
-                        vc.validateAlternateAlleles();
-                        vc.validateChromosomeCounts();
-                    }
-                    else{
-                        vc.validateAlternateAlleles();
-                        vc.validateChromosomeCounts();
-                        vc.validateRSIDs(rsIDs);
-                    }
-                }
-                break;
-            case REF:
-                vc.validateReferenceBases(reportedRefAllele, observedRefAllele);
-                break;
-            case IDS:
-                if (!rsIDs.isEmpty()) {
-                    vc.validateRSIDs(rsIDs);
-                }
-                break;
-            case ALLELES:
-                vc.validateAlternateAlleles();
-                break;
-            case CHR_COUNTS:
-                vc.validateChromosomeCounts();
+       if(validationsToPerform[ValidationType.REF.ordinal()]) {
```

Comment:
Considering that this method has grown in complexity now that it is handling everything, I would advocate all of these validations either be pulled out into their own sub methods or that we add comments above each of the cases here explaining what each annotation is doing.